app/models/message.rb

```diff
@@ -0,0 +1,23 @@
+# == Schema Information
+#
+# Table name: messages
+#
+#  id              :bigint           not null, primary key
+#  text            :text             not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  conversation_id :bigint           not null
+#  user_id         :bigint
+#
+# Indexes
+#
+#  index_messages_on_conversation_id  (conversation_id)
+#  index_messages_on_user_id          (user_id)
+#
+
+class Message < ApplicationRecord
+  belongs_to :user, optional: true
```

Comment:
Why is this optional needed? Does make sense a message without a user? (Maybe yes)

Reply:
When a there is a target match, a new message is sent in the conversation saying 'new target match' ; you can match several times and since you only have one conversation you will see as many messages as matches. This messages dont have a user.